test/test_testing.py

```diff
@@ -1881,5 +1881,35 @@ def test_sample_input_metadata(self) -> None:
         self.assertEqual(s2.name, "foo")
 
 
+class TestOpInfoSampleFunctions(TestCase):
+
+    def _assert_is_generator_or_singleton(self, item, property_name):
+        if isinstance(item, Sequence):
+            msg = (
+                "{property_name} may only return lists for single items"
+                ", please use a coroutine which yields items instead")
+            self.assertTrue(len(item) <= 1, msg=msg)
+        else:
+            self.assertIsInstance(item, Generator)
+
+    @ops(op_db, dtypes=OpDTypes.any_one)
```

Comment:
These tests make a lot of sense (comments describing each one, please!) 

Should we consider adding these checks to the tests in `test_ops.py` that exercise these generators, instead? That's not conceptually ideal because then those tests are testing multiple things (they probably already are), but it would reduce duplicative work in the test suite, which does a lot already. Or are these new tests so fast we shouldn't be concerned about their speed?

An example of a similar case is the test for "input" tensors having no grad history, which is in the noncontiguous samples test:

https://github.com/pytorch/pytorch/blob/c27a5171b80b126917ef4435e232055415fcf617/test/test_ops.py#L503

cc @lezcano

Reply:
A great thing to have would be a test that tests OpInfo samples themselves, and have all these tests grouped there. Until then, if this test is fast enough, it may be fine.

Reply:
On my machine 
```
time pytest test_testing.py -k TestOpInfoSampleFunctions
```
takes less than 10 seconds even on a build with CUDA. So I'm not particularly concerned.

On the other hand if we combine these checks into an existing test, then the test is more likely to be skipped for no good reason.

Reply:
SGTM